### PR TITLE
Upgrade react to 16.8.2.

### PR DESCRIPTION
**Pull request by mattias800**
Upgrade to emotion 10.
Move common libraries to peerDependencies.
Update README.
Fix a few type errors.
Update ArrowBox to use emotions styled API instead of css API.

### CODE REVIEW

**Comment by mattias800** [comment]
`Move common libraries to peerDependencies` .. This is no longer included in this PR.

File: src/components/ui/progress/BoatWrapper.ts [hunk 1]
```diff
@@ -72,29 +74,29 @@ export const boatProgressIndicatorStyle = css(`
...
     background-size: 133% 144%;
     -webkit-animation: sail 4s linear infinite;
     animation: sail 4s linear infinite;
-}
+  }
 
-@-webkit-keyframes sail {
+  @-webkit-keyframes sail {
```
Comment (lindskogen) [inline src/components/ui/progress/BoatWrapper.ts]: Not sure we need prefixed CSS? Doesn't emotion use autoprefixer?

File: src/components/ui/interaction/Clickable.tsx [hunk 2]
```diff
@@ -60,8 +64,9 @@ export class ClickableComponent extends React.Component<ClickableInnerProps> {
     const hasClickHandler = !!(onClick || onDblClick);
     const opacity = !disableOpacityOnClick && mouseIsDown ? 0.5 : undefined;
     return (
-      <div
-        className={opacityOnHover ? hoverStyle : undefined}
+      <
+
```
Comment (lindskogen) [inline src/components/ui/interaction/Clickable.tsx]: What happened here? No `div`?

File: src/components/ui/progress/BoatWrapper.ts (hunk 1 above)
Comment (lindskogen) [reply src/components/ui/progress/BoatWrapper.ts]: I checked, no need for vendor-prefixes :)

Comment (mattias800) [reply src/components/ui/progress/BoatWrapper.ts]: Like I said, it's copy paste CSS from some generator :)

File: src/components/ui/interaction/Clickable.tsx (hunk 2 above)
Comment (mattias800) [reply src/components/ui/interaction/Clickable.tsx]: All usage of className and `css` has been replaced with `styled`.

Comment (mattias800) [reply src/components/ui/interaction/Clickable.tsx]: Not a huge fan of the resulting code.

```
const HoverStyledDiv = setDisplayName<
  HoverStyledDivProps &
    React.DetailedHTMLProps<
      React.HTMLAttributes<HTMLDivElement>,
      HTMLDivElement
    >
>('HoverStyledDiv')(styled.div<HoverStyledDivProps>`
  :hover {
    ${props => (props.opacityOnHover ? 'opacity: 0.7;' : '')};
  }
`);
```

Comment (mattias800) [reply src/components/ui/interaction/Clickable.tsx]: Fixed :)

File: src/components/ui/progress/BoatWrapper.ts (hunk 1 above)
Comment (mattias800) [reply src/components/ui/progress/BoatWrapper.ts]: Fixed!

**Pull request merged**